pkg/components/pubsub/pluggable.go

```diff
@@ -0,0 +1,210 @@
+/*
+Copyright 2022 The Dapr Authors
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package pubsub
+
+import (
+	"context"
+	"io"
+	"sync"
+
+	"github.com/dapr/components-contrib/pubsub"
+
+	"github.com/dapr/dapr/pkg/components"
+	"github.com/dapr/dapr/pkg/components/pluggable"
+	proto "github.com/dapr/dapr/pkg/proto/components/v1"
+
+	"github.com/dapr/kit/logger"
+
+	"github.com/pkg/errors"
+)
+
+// grpcPubSub is a implementation of a pubsub over a gRPC Protocol.
+type grpcPubSub struct {
+	*pluggable.GRPCConnector[proto.PubSubClient]
+	// features the list of state store implemented features features.
+	features []pubsub.Feature
+	logger   logger.Logger
+}
+
+// Init initializes the grpc pubsub passing out the metadata to the grpc component.
+// It also fetches and set the component features.
+func (p *grpcPubSub) Init(metadata pubsub.Metadata) error {
+	if err := p.Dial(metadata.Name); err != nil {
+		return err
+	}
+
+	protoMetadata := &proto.MetadataRequest{
+		Properties: metadata.Properties,
+	}
+
+	// TODO Static data could be retrieved in another way, a necessary discussion should start soon.
+	// we need to call the method here because features could return an error and the features interface doesn't support errors
+	featureResponse, err := p.Client.Features(p.Context, &proto.FeaturesRequest{})
```

Comment:
Should features call be made only after init is successful. Again I understand that static features does not necessarily need init first. But should we replicate the flow that runtime is following for now for all components?

Reply:
That's a good expectation. I agree with you. Thanks

My hope is that this line should be deleted soon